lightning/src/ln/blinded_payment_tests.rs

```diff
@@ -2081,29 +2088,44 @@ fn do_test_trampoline_single_hop_receive(success: bool) {
 
 	check_added_monitors!(&nodes[0], 1);
 
-	if success {
+	if matches!(scenario, TrampolineSingleHopReceiveCheck::SuccessfulBlindedReceive) {
```

Comment:
Hmm yes the `scenario` value seems to push this to the point where it becomes hard to read. I'd say: just acceptable? There are probably other ways to compose a test from reusable building blocks, but maybe not worth it now.

Reply:
Up to you.